cdap-ui/app/directives/log-viewer/log-viewer.js

```diff
@@ -270,6 +271,49 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi, LOGVIEWERSTORE_
     });
   };
 
+  var exportTimeout = null;
+
+  const downloadLogs = () => {
+    return myLogsApi.getLogsStartAsRaw({
+      namespace : this.namespaceId,
+      appId : this.appId,
+      programType : this.programType,
+      programId : this.programId,
+      runId : this.runId,
+      start : this.startTimeSec
+    }).$promise.then(
+    (res) => {
+      this.downloadContent = res;
+    },
+    (err) => {
+      console.log('ERROR: ', err);
+    });
+  };
+
+  this.export = () => {
+    downloadLogs().then( () => {
+      var blob = new Blob([this.downloadContent], {type: 'text/plain'});
+        this.url = URL.createObjectURL(blob);
+        let filename = '';
+        if ('undefined' !== typeof this.getDownloadFilename()) {
+          filename = this.getDownloadFilename() + '-' + this.startTimeSec;
+        } else {
+          filename = this.namespaceId + '-' + this.appId + '-' + this.programId + '-' + this.startTimeSec;
```

Comment:
The star time is in epoch format. I am not sure if that will be user friendly though..